StudyHub/StudyHub.BLL/Services/Auth/GoogleAuthService.cs

```diff
@@ -1,45 +1,71 @@
 ﻿using Google.Apis.Auth.OAuth2.Flows;
 using Google.Apis.Auth.OAuth2;
 using Google.Apis.Auth;
-using StudyHub.BLL.Services.Interfaces;
 using StudyHub.Common.DTO.AuthDTO;
 using StudyHub.Common.Exceptions;
 using StudyHub.Entities;
 using Microsoft.AspNetCore.Identity;
 using StudyHub.Common.Models;
 using Microsoft.Extensions.Options;
+using StudyHub.BLL.Services.Interfaces.Auth;
+using AutoMapper;
 
-namespace StudyHub.BLL.Services;
+namespace StudyHub.BLL.Services.Auth;
 
 public class GoogleAuthService : IGoogleAuthService
 {
     private readonly UserManager<User> _userManager;
     private readonly ITokenService _tokenService;
     private readonly GoogleAuthConfig _googleConfig;
+    private readonly IMapper _mapper;
 
     public GoogleAuthService(
         UserManager<User> userManager,
         ITokenService tokenService,
-        IOptions<GoogleAuthConfig> googleConfig)
+        IOptions<GoogleAuthConfig> googleConfig,
+        IMapper mapper)
     {
         _userManager = userManager;
         _tokenService = tokenService;
         _googleConfig = googleConfig.Value;
+        _mapper = mapper;
     }
-    public async Task<AuthSuccessDTO> GoogleLogin(string authorizationCode)
+
+    public async Task<AuthSuccessDTO> GoogleRegisterAsync(string authorizationCode)
+    {
+        var payload = await GetGooglePayloadAsync(authorizationCode);
+
+        var user = await _userManager.FindByEmailAsync(payload.Email);
+
+        if (user is not null)
```

Comment:
code style. I mean, meaby you make this part same like in other files to save identity of code style?